### DISCUSSION TITLE
not send caption with file id in photo/doc

### DISCUSSION THREAD

**Discussion by ah-rahimi** [def-studio/telegraph#156] [General]
on method photo()/socument() can't send with file id
```$this->user->photo($cover->photo_id)->send()```

**Comment by fabio-ivona** [comment]
do you have any exception?

**Reply by ah-rahimi** [reply]
I apologize, the problem was with my system

**Comment by ah-rahimi** [comment]
But I am sure that the html method does not work in the following statement and the caption is not sent: @fabio-ivona 

![image](https://user-images.githubusercontent.com/87445571/186654228-2a1cd131-1d40-4541-8e0c-9cfb3a336e1c.png)

**Reply by fabio-ivona** [reply]
@ahr1384 it seems to work as well

```php
Telegraph::document(Storage::path('test.txt'))->html('Hi')->send()
```


![image](https://user-images.githubusercontent.com/8792274/186655206-04e997ae-5d6e-42f5-89f0-552640ef583e.png)


try to dump the result with `->send()->dump()`

**Reply by ah-rahimi** [reply]
![image](https://user-images.githubusercontent.com/87445571/186656087-00a3d4ca-ef45-4395-b449-54758cccb91d.png)
No, it will not be sent again

**Reply by ah-rahimi** [reply]
Maybe there is a problem in sending with file id

**Reply by ah-rahimi** [reply]
yes i test it @fabio-ivona
![image](https://user-images.githubusercontent.com/87445571/186656488-5fe46d47-b3ac-43a9-8861-21b2dcc3bb75.png)

There is a problem with the file id when sending

**Reply by fabio-ivona** [reply]
nope, it seems to work:

```php
$response = Telegraph::document(Storage::path('test.txt'))->message('first')->send();

$fileId = $response->json('result.document.file_id');

Telegraph::document($fileId)->message('resend')->send();
```

![image](https://user-images.githubusercontent.com/8792274/186692334-824ed7c5-f3fe-4386-ba7a-17117edca12d.png)

**Reply by ah-rahimi** [reply]
i try with html() but you try with message()

**Reply by fabio-ivona** [reply]
same result:

![image](https://user-images.githubusercontent.com/8792274/186693582-bd8be510-46ce-4d3e-b555-c1a01657eb57.png)


![image](https://user-images.githubusercontent.com/8792274/186693640-7cc27374-2bec-4946-8390-6383abb035f8.png)

**Reply by ah-rahimi** [reply]
But message() doesn't work either

**Reply by ah-rahimi** [reply]
I don't know where the problem is
But when I try to send with file ID, it doesn't come
It comes with a link

**Reply by ah-rahimi** [reply]
please check in $this->chat no Telegraph::

**Reply by fabio-ivona** [reply]
please, see the result of your 

->send()->dd();

it should tell you where the problem is

**Reply by ah-rahimi** [reply]
whattttt

this is not work: 
![image](https://user-images.githubusercontent.com/87445571/186696888-30e857b2-8718-4b5d-b72f-a25db1a6e5c1.png)

but this is work:
![image](https://user-images.githubusercontent.com/87445571/186696971-8d7bad5d-aeaf-4bd7-8d1f-99b8c74f3478.png)


**I do not understand**

**Reply by ah-rahimi** [reply]
I don't understand what it has to do with it
But if something is sent before it, it will work properly

Please copy your file id and send it in a separate request
Not after sending the file

**Reply by fabio-ivona** [reply]
it works:

![image](https://user-images.githubusercontent.com/8792274/186699957-0fc70d24-364b-45b6-a208-a9a2a6fee6a1.png)

![image](https://user-images.githubusercontent.com/8792274/186700006-fef2fb63-8f3b-4848-9378-980c733d3dde.png)

**Reply by fabio-ivona** [reply]
please print here the result of the send()->dd()

**Reply by ah-rahimi** [reply]
![image](https://user-images.githubusercontent.com/87445571/186701215-62bf7ce2-cc97-4253-b418-af70bb269908.png)

for this:

![image](https://user-images.githubusercontent.com/87445571/186701239-5b0f21ba-2d8b-400d-b4bc-b559e2b3274e.png)

**Reply by fabio-ivona** [reply]
ok, found the issue, am releasing a fix

**Reply by ah-rahimi** [reply]
thanks

**Comment by ah-rahimi** [comment]
I just realized that we have functions called dump() and dd()
Please put it in the document for others to use

**Reply by fabio-ivona** [reply]
done
